pycc/ccresponse.py

```diff
@@ -391,14 +504,310 @@ def r_X2(self, pertbar, omega):
 
         return r_X2
 
+    def in_Y1(self, pertbar, X1, X2):
```

Comment:
I don't understand the `in_Y1()`/`r_Y1()` and `in_Y2()/r_Y2()` combinations.  Why can't we just have `r_Y1()` and `r_Y2()` in analogy to the corresponding right-hand perturbed wave function residuals?

Update: Wait, maybe I do.  Are these the components of the `r_Y1` and `r_Y2` residuals that aren't dependent on `Y1` and `Y2` so you only want to compute them once?

Reply:
Yes